lib/Interpreter/IncrementalParser.cpp

```diff
@@ -125,7 +125,7 @@ namespace {
   if (!II)
     return;
   const clang::DefMacroDirective* MD
-    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
+    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
```

Comment:
On 15/06/16 19:06, marsupial wrote:
>
> In lib/Interpreter/IncrementalParser.cpp 
> <https://github.com/root-mirror/cling/pull/17#discussion_r67203107>:
>
> > @@ -125,7 +125,7 @@ namespace {
> >    if (!II)
> >      return;
> >    const clang::DefMacroDirective* MD
> > -    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
> > +    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
>
> Yes, that is the issue, but code later is prepared to handle that. The 
> problem is llvm:: dyn_cast cannot take a null value.
> If PP.getMacroDirective(II) returns dyn_cast_or_null just propagates 
> it to MD, which is checked for NULL on the next line.
>
I think this is hiding the real issue. PP.getMacroDirective(II) must not 
return null in this case. Could you give us the setup in which that 
happened?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/root-mirror/cling/pull/17/files/252d215c9d8be48c614e85447f52951f37135ec5..6ac37ef57b76f3b4469856f12ce069b03a57c678#r67203107>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AGNuU7xMxzmgUMCOm9QkdFE3_wfo-HToks5qMDEtgaJpZM4I0etZ>.
>